payroll/v_payroll.payroll_audit.sql

```diff
@@ -0,0 +1,353 @@
+             ,CONVERT(NVARCHAR(16), ern_3_ret) AS ern_3_ret
+             ,CONVERT(NVARCHAR(16), ern_4_ac) AS ern_4_ac
+             ,CONVERT(NVARCHAR(16), ern_4_ba) AS ern_4_ba
+             ,CONVERT(NVARCHAR(16), ern_4_bon) AS ern_4_bon
+             ,CONVERT(NVARCHAR(16), ern_4_db) AS ern_4_db
+             ,CONVERT(NVARCHAR(16), ern_4_dc) AS ern_4_dc
+             ,CONVERT(NVARCHAR(16), ern_4_dei) AS ern_4_dei
+             ,CONVERT(NVARCHAR(16), ern_4_ds) AS ern_4_ds
+             ,CONVERT(NVARCHAR(16), ern_4_ecl) AS ern_4_ecl
+             ,CONVERT(NVARCHAR(16), ern_4_emh) AS ern_4_emh
+             ,CONVERT(NVARCHAR(16), ern_4_epc) AS ern_4_epc
+             ,CONVERT(NVARCHAR(16), ern_4_glc) AS ern_4_glc
+             ,CONVERT(NVARCHAR(16), ern_4_hb) AS ern_4_hb
+             ,CONVERT(NVARCHAR(16), ern_4_ic) AS ern_4_ic
+             ,CONVERT(NVARCHAR(16), ern_4_lp) AS ern_4_lp
+             ,CONVERT(NVARCHAR(16), ern_4_ntp) AS ern_4_ntp
+             ,CONVERT(NVARCHAR(16), ern_4_oob) AS ern_4_oob
+             ,CONVERT(NVARCHAR(16), ern_4_ret) AS ern_4_ret
+             ,CONVERT(NVARCHAR(16), ern_4_trs) AS ern_4_trs
+             ,CONVERT(NVARCHAR(16), ern_5_ret) AS ern_5_ret
+             ,CONVERT(NVARCHAR(16), fli_amount) AS fli_amount
+             ,CONVERT(NVARCHAR(16), hrs_4_bev) AS hrs_4_bev
+             ,CONVERT(NVARCHAR(16), hrs_4_cvd) AS hrs_4_cvd
+             ,CONVERT(NVARCHAR(16), hrs_4_prd) AS hrs_4_prd
+             ,CONVERT(NVARCHAR(16), hrs_4_pto) AS hrs_4_pto
+             ,CONVERT(NVARCHAR(16), hrs_4_rel) AS hrs_4_rel
+             ,CONVERT(NVARCHAR(16), hrs_4_sic) AS hrs_4_sic
+             ,CONVERT(NVARCHAR(16), local_cd_1) AS local_cd_1
+             ,CONVERT(NVARCHAR(16), local_cd_2) AS local_cd_2
+             ,CONVERT(NVARCHAR(16), local_tax_1) AS local_tax_1
+             ,CONVERT(NVARCHAR(16), local_tax_2) AS local_tax_2
+             ,CONVERT(NVARCHAR(16), med_surtax) AS med_surtax
+             ,CONVERT(NVARCHAR(16), med_tax) AS med_tax
+             ,CONVERT(NVARCHAR(16), ot_ern) AS ot_ern
+             ,CONVERT(NVARCHAR(16), ot_hrs) AS ot_hrs
+             ,CONVERT(NVARCHAR(16), rate) AS rate
+             ,CONVERT(NVARCHAR(16), rate_2) AS rate_2
+             ,CONVERT(NVARCHAR(16), rate_3) AS rate_3
+             ,CONVERT(NVARCHAR(16), rate_used) AS rate_used
+             ,CONVERT(NVARCHAR(16), rg_ern) AS rg_ern
+             ,CONVERT(NVARCHAR(16), rg_hrs) AS rg_hrs
+             ,CONVERT(NVARCHAR(16), sdi_tax) AS sdi_tax
+             ,CONVERT(NVARCHAR(16), ss_tax) AS ss_tax
+             ,CONVERT(NVARCHAR(16), state_tax_1) AS state_tax_1
+             ,CONVERT(NVARCHAR(16), state_tax_2) AS state_tax_2
+             ,CONVERT(NVARCHAR(16), sui_tax) AS sui_tax
+             ,CONVERT(NVARCHAR(16), n_records) AS n_records
+
+       FROM payroll_rollup r
+       LEFT JOIN (SELECT DISTINCT employee_number, position_id FROM gabby.people.employment_history) eh
```

Comment:
remove `JOIN`, you're pulling employee_number below

Reply:
makes sense

Reply:
oh wait! I'm using employee number on the lag functions in the payroll_unpivot subquery, rows 152 and 153. If I were to use position_id there instead of employee_number, the LAG function wouldn't connect the most recent payroll information for someone who changes legal entities. Those are folks we definitely want to include in the audit. I'm happy to adjust if there's a better way to get a clean crosswalk between all the position_id and employee_number combinartions.